torch/_dynamo/variables/functions.py

```diff
@@ -691,11 +692,27 @@ def call_function(
         # call_function must check any unsupported arguments and graph-break.
         # It's safe to assume args/kwargs from orig_fn map 1:1 to args/kwargs of remapped_fn,
         # since that's the contract for putting a mapping in `traceable_collective_remaps`
-        if "async_op" in kwargs and kwargs["async_op"].as_python_constant():
+
+        # Merge args and kwargs so positional and keyword args
+        # can be located the same way.
+        signature = inspect.signature(self.fn)
+        arg_names = list(signature.parameters.keys())
+        new_kwargs = {}
+        for i in range(len(arg_names)):
+            if i < len(args):
```

Comment:
Doing the manual args/kwargs handling seems open to subtle errors. Is there any way to make python bind the given args/kwargs to a function for you and then use that as the source of truth?

Reply:
Good point. I think Signature.bind would do the trick.